src/client/model/dalNamespace.ts

```diff
@@ -0,0 +1,76 @@
+import { ElectricNamespace } from '../../electric/namespace'
+import { DatabaseAdapter } from '../../electric/adapter'
+import { Notifier } from '../../notifiers'
+import { DBDescription, TableDescription } from './dbDescription'
+import { Table } from './table'
+
+export type DalTables<DB extends DBDescription<any>> = {
+  [Tbl in keyof DB['tables']]: DB['tables'][Tbl] extends TableDescription<
+    infer T,
+    infer CreateData,
+    infer UpdateData,
+    infer Select,
+    infer Where,
+    infer WhereUnique,
+    infer Include,
+    infer OrderBy,
+    infer ScalarFieldEnum,
+    infer GetPayload
+  >
+    ? Table<
+        T,
+        CreateData,
+        UpdateData,
+        Select,
+        Where,
+        WhereUnique,
+        Include,
+        OrderBy,
+        ScalarFieldEnum,
+        GetPayload
+      >
+    : never
+}
+
+// Extends the ElectricNamespace with a `db` property that is the client of the data access library
+export class DalNamespace<
```

Comment:
Yes, that was pretty bad naming. Renamed it to `Electric`.
Since it is parametric it is a bit odd that programmers need to write `Electric<typeof dbDescription>`. Perhaps we can let the generator generate 2 more types:
```
export type DbDescription = typeof dbDescription
export type ElectricClient = Electric<DbDescription>
```

Reply:
Nice :)

Do you definitely like "description" as opposed to "schema"? I'd never talk about a DB description but I'd often talk about a DB schema.

Reply:
Sure, will call it schema then :-)

Reply:
🙏